mongo/errors.go

```diff
@@ -71,6 +73,56 @@ func replaceErrors(err error) error {
 	return err
 }
 
+// IsDuplicateKeyError returns true if err is a duplicate key error
+func IsDuplicateKeyError(err error) bool {
+	// handles SERVER-7164 and SERVER-11493
+	if e, ok := err.(ServerError); ok {
```

Comment:
Should this also be inspecting any possible nested errors on `err`, like `IsTimeout` currently does?